python/analysis/auto_generated/interpolation/qgsaidwinterpolation.sip.in

```diff
@@ -0,0 +1,50 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/analysis/interpolation/qgsaidwinterpolation.h                    *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+
+class QgsAidwInterpolation
+{
+%Docstring(signature="appended")
+The :py:class:`QgsAidwInterpolation` class implements OpenCL accelerated IDW interpolation.
+This implementation requires OpenCL, if QGIS is build without OpenCL support the algorithm
+throws an exception.
+
+.. versionadded:: 3.22
+%End
+
+%TypeHeaderCode
+#include "qgsaidwinterpolation.h"
+%End
+  public:
+
+    QgsAidwInterpolation( QgsVectorLayer *dataLayer, QString &dataAttributeName, QgsRasterLayer *interpolatedLayer, double coefficient = 2.0 );
+%Docstring
+Constructs a QgsAidwInterpolation object.
+
+:param dataLayer: the input points layer.
+:param dataAttributeName: name of the input layer attribute that contains the values for the interpolation, it must be a numeric field.
+:param interpolatedLayer: the destination layer for the interpolation, data type must be Float64 (double precision).
+:param coefficient: coefficient for distance weighting, default to 2.0.
+%End
+
+    void process( QgsFeedback *feedback = 0 ) throw( QgsProcessingException );
```

Comment:
Is QgsProcessingException the correct exception here? This class isn't processing specific, so I'd argue that QgsNotSupportedException is more appropriate